packages/store/slices/dapps-store.ts

```diff
@@ -0,0 +1,74 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import { getSelectedApps } from "../../screens/DAppStore/query/query";
+import { dAppGroup } from "../../screens/DAppStore/types";
+import { RootState } from "../store";
+
+type pinnedAppsCollection = string[];
+
+interface DappsStorage {
+  selectedApps: pinnedAppsCollection;
+  availableApps: dAppGroup;
+}
+
+const initialState: DappsStorage = {
+  selectedApps: getSelectedApps(),
+  availableApps: {},
+};
+
+export const selectAvailableApps = (state: RootState) =>
+  state.dAppsStore.availableApps;
+
+export const selectCheckedApps = (state: RootState) => [
+  ...new Set(state.dAppsStore.selectedApps),
+];
+
+const persist = (state: pinnedAppsCollection) => {
+  state = [...new Set(state)];
+  const pureMagic = JSON.stringify(state, function replacer(key, value) {
+    if (Array.isArray(value) && value.length === 0) {
+      return { ...value }; // Converts empty array with string properties into a POJO
+    }
+    return value;
+  });
+  window.localStorage.setItem("teritori-dappstore", pureMagic);
+};
```

Comment:
there is already a persistence system as redux middleware, you only need to whitelist your slice

Reply:
resolved with [ba6b190](https://github.com/TERITORI/teritori-dapp/pull/401/commits/ba6b1900e6c8aff2c30d0c67c9820b3ba54e83da)